### ISSUE TITLE
[QUESTION] Quest tracking

### ISSUE THREAD

**Issue by winterismute** [HearthSim/Hearthstone-Deck-Tracker#3011]
Hi, I found out about this project fairly recently, I am not an HS powerplayer but I am still playing sometimes and I'm finding this SW not only useful but also well done, great job!
A question: I am interested in tracking quests (given, progress, completed etc), if I am not terribly wrong it seems it's not being done ATM, hence my questions:
- Is anybody planning to add it or it's not a priority?
- To implement it, where would you start? From a quick look at the code it seems all the event tracking is done via watching log files, but from a first glance I could not find any log file containing info about quests...

**Comment by azeier** [comment]
I don't think this is possible without basically re-implementing all the quest tracking logic, so this is likely not a feasible feature.

**Comment by winterismute** [comment]
But is there any quest tracking being done at all?

**Comment by azeier** [comment]
In HDT? no

**Comment by jleclanche** [comment]
@winterismute I think this would fit perfectly in a plugin.

**Comment by winterismute** [comment]
True! My second question was whether you know if watching a log file might be enough to intercept those events, I could not find anything about quests on the log files but maybe it's just me...

**Comment by jleclanche** [comment]
Logfiles won't be enough. You'll need to hook into HearthMirror functionality: https://github.com/HearthSim/HearthMirror

**Comment by jleclanche** [comment]
Also, you'll probably need to add that functionality to HM ;)

**Comment by culinko** [comment]
Some logging has been added to Achievements.log recently, because today I replaced one of my quests and this is how it looked:
```
D 17:27:23.2322361 Blocking AchievementNotification: ID=609
D 17:27:23.2322361 Blocking AchievementNotification: ID=600
D 17:27:23.2342361 PopupDisplayManager: adding 0 rewards to load total=0
D 17:27:23.3292397 OnQuestCanceled: CanceledQuest=[CanceledQuest AchieveID=609 Canceled=True NextQuestCancelDate=131254560000000000]
D 17:27:23.3322398 OnAchieveNotification PlayerID=652227 ID=609 Complete=False New=False Remove=True Amount=0
D 17:27:23.3332398 OnAchievementNotification: Achievement=[Achievement: ID=609 AchieveGroup=DAILY_QUEST Name='Mrgrgrglrr' MaxProgress=30 Progress=0 AckProgress=0 IsActive=False DateGiven=131223397330000000 DateCompleted=131223397330000000 Description='Play 30 Murlocs.' Trigger=IGNORE CanAck=True]
D 17:27:23.3332398 OnAchieveNotification PlayerID=652227 ID=600 Complete=False New=True Remove=False Amount=0
D 17:27:23.3332398 OnAchievementNotification: Achievement=[Achievement: ID=600 AchieveGroup=DAILY_QUEST Name='Mage Mastery' MaxProgress=50 Progress=0 AckProgress=-1 IsActive=True DateGiven=131254288430000000 DateCompleted=0 Description='Play 50 Mage class cards.' Trigger=IGNORE CanAck=False]
D 17:27:23.3342399 PopupDisplayManager: adding 0 rewards to load total=0
D 17:27:23.3372400 Adding QuestLog tile for: [Achievement: ID=600 AchieveGroup=DAILY_QUEST Name='Mage Mastery' MaxProgress=50 Progress=0 AckProgress=-1 IsActive=True DateGiven=131254288430000000 DateCompleted=0 Description='Play 50 Mage class cards.' Trigger=IGNORE CanAck=False]
```

**Comment by jleclanche** [comment]
@culinko did you post a log? if there is an event in the log then that's pretty useful to know to trigger a memory scan.

Also where have you been dude? :)

**Comment by culinko** [comment]
Yeah I posted a part of the log, but later realized that it was only triggered by me rerolling a quest (it showed old and new quest in Achievements.log). Other quests were not displayed there at all, that's why I later removed my comment.

It was a combination of me being busy/lost interest in the game that kept me away from HS. I'm still around, checking the mails I get from issues I'm subscribed to and today I just checked out what's new in HDT/hs-bugs :)

**Comment by jleclanche** [comment]
Reroll triggers are still pretty useful, as this would be a needed feature in a quest tracker.

Also, we're mirroring the IRC channel [on discord](https://hearthsim.info/join/) now, so you won't have an excuse to be MIA anymore ;)

**Comment by winterismute** [comment]
> Yeah I posted a part of the log, but later realized that it was only triggered by me rerolling a quest (it 
> showed an old and a new quest in Achievements.log). Other quests were not displayed there at all, 
> that's why I later removed my comment. Also Ben.log mentions how many active quests you have and 
> updates every time you open the quest window.

Well, if the same event is triggered when a NEW quest is gained, then that's it :D Will try now.

@jleclanche is there any document that could help me understand what HearthMirror does? Reading from system memory from another process is something kind of new for me...

**Comment by jleclanche** [comment]
You can ask for help on our [discord server](https://hearthsim.info/join/). Just ping Alex there.

**Comment by riQQ** [comment]
There is https://github.com/arkein/Hearthstone-Treasury.

**Comment by MGK82** [comment]
I think that's not what was originally asked for.

As far as I know, quest progress is logged in the achievement.log file.

I would look for that. But at the moment I am busy with the Pack Tracker plugin. So, if anybody feels like it, go for it.

**Comment by riQQ** [comment]
I haven't tried it, but it should at least track quest completion when you're rewarded gold.

**Comment by MGK82** [comment]
Today, I was up early and played a bit before work. The log file is still there, having the following entries (in ger tho):
`D 07:44:33.5284312 OnAchievementNotification: Achievement=[Achievement: ID=47 AchieveGroup=DAILY_QUEST Name='Dominanz des Priesters oder Hexenmeisters' MaxProgress=5 Progress=0 AckProgress=-1 IsActive=True DateGiven=131455214730000000 DateCompleted=131445879460000000 Description='Gewinnt 5 Spiele als Priester oder Hexenmeister.' Trigger=WIN_GAME CanAck=True]`

`D 07:55:15.5131506 OnAchievementNotification: Achievement=[Achievement: ID=47 AchieveGroup=DAILY_QUEST Name='Dominanz des Priesters oder Hexenmeisters' MaxProgress=5 Progress=1 AckProgress=0 IsActive=True DateGiven=131455214730000000 DateCompleted=131445879460000000 Description='Gewinnt 5 Spiele als Priester oder Hexenmeister.' Trigger=WIN_GAME CanAck=True]`

And so on

**Comment by winterismute** [comment]
Hi,
Just wanted to say that, after I started this issue, I actually tried to implement the task tracking from the logfiles myself, at this branch (example commit): https://github.com/winterismute/Hearthstone-Deck-Tracker/commit/018f6707f34f498356d343881b592687b98bd9af
I did have some issues with I think it was sort of working. Unfortunately, I lost interest in Hearthstone as a game pretty soon afterwards...

**Comment by rbrasga** [comment]
Hello @winterismute, I'm interested in picking this up.

Have any tips on getting started? Apparently the discord is good place to coordinate efforts and ask questions. Anything else I should know?

**Comment by ksurl** [comment]
I found [this](https://github.com/PyroGenesis/Hearthstone-Quest-Tracker). it needs to be updated for monster hunt and the reworked quest rewards.
